lib/metadataManager.ts

```diff
@@ -0,0 +1,131 @@
+import { Client } from "urql";
+
+import {
+  FetchAppDetailsDocument,
+  FetchAppDetailsQuery,
+  UpdateAppMetadataDocument,
+} from "../generated/graphql";
+
+const debug = console.error;
+
+export type SettingsValueWithEnv = {
+  key: string;
+  value: string;
+  domain?: string;
+};
+
+export interface SettingsManagerInterface {
+  get: (key: string, domain?: string) => Promise<string | undefined>;
+  set: (settings: SettingsValueWithEnv[]) => Promise<void>;
+}
+
+const deserializeMetadata = (key: string, value: string): SettingsValueWithEnv => {
+  // domain specific metadata use convention key__domain, e.g. `secret_key__example.com`
```

Comment:
As agreed - your solution is fine. We can stringify/parse values for users too, but if we make it explicit (in docs) that value must be stringified first - its probably fine. But IMO from DX perspective it can be part of sdk